horizomer/utils/tree.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+
+# ----------------------------------------------------------------------------
+# Copyright (c) 2015--, The Horizomer Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+#
+# functions relevant to phylogenetic tree operations
+#
+
+from skbio import TreeNode
+
+
+def compare_topology(node1, node2):
+    """Test whether the topologies of two trees are identical.
+
+    Parameters
+    ----------
+    tree1 : skbio.TreeNode
+        first tree in comparison
+    tree2 : skbio.TreeNode
+        second tree in comparison
+
+    Returns
+    -------
+    bool
+        whether the topologies are identical
+
+    Notes
+    -----
+    Flipping child nodes does not change topology.
+    Branch lengths are ignored.
+    """
+    if len(node1.children) != len(node2.children):
+        return False
+    elif len(node1.children) > 0:
+        # check if sorted child names are identical
+        children1 = sorted(node1.children, key=lambda x: x.name)
+        children2 = sorted(node2.children, key=lambda x: x.name)
+        taxa1 = [x.name for x in children1]
+        taxa2 = [x.name for x in children2]
+        if taxa1 != taxa2:
+            return False
+
+        # recursively test all children
+        res = []
+        for child1, child2 in zip(children1, children2):
+            res.append(compare_topology(child1, child2))
```

Comment:
uh oh,  looks like there is a bit of recursion here.  This will not play well with larger tree, as the Python stack will need to be continuously adjusted to handle deep recursion.

I'd suggest swapping this out with a [pre-order traversal](http://scikit-bio.org/docs/latest/generated/skbio.tree.TreeNode.preorder.html#skbio.tree.TreeNode.preorder).  Alternatively we can raise an issue to tackle this later.

Reply:
This sounds a good idea. However I just tested and realized that preorder traversal doesn't consider the order of child nodes under one parent node. For example, I did:
```
def compare_topology(tree1, tree2):
    for (node1, node2) in zip(tree1.preorder(), tree2.preorder()):
        if node1.name != node2.name:
            return False
        children1 = sorted(node1.children, key=lambda x: x.name)
        children2 = sorted(node2.children, key=lambda x: x.name)
        taxa1 = [x.name for x in children1]
        taxa2 = [x.name for x in children2]
        if taxa1 != taxa2:
            return False
    return True
```
Then I tested it:
```
tree1 = '(a,b)c;'
tree2 = '(b,a)c;'
compare_topology(TreeNode.read([tree1]), TreeNode.read([tree2]))
```
The answer is False. However they actually should be identical, since `a` and `b` are merely flipped.